docker/build.sh

```diff
@@ -1,15 +1,59 @@
 #! /usr/bin/env bash
 set -euo pipefail
 
+# 1. pass the desired version of CUDA to build.sh 
+# (if not, build with the version on current machine, produce an error if no CUDA installed)
+# 2. pass the CUDA version to docker file and use CUDA images based on input version
+# 3. Add the CUDA version as a suffix to the image name (both for the environment and flexflow/specinfer images).
+# 4. Edit the publish.sh / workflow file to publish one image per version, using for example 11.1, 11.3, 11.5, 11.6, 11.7, 11.8 (you can start by just building only two versions, e.g. 11.1 and 11.8 to check if it works)
+
 # Usage: ./build.sh <docker_image_name>
 
 # https://stackoverflow.com/questions/59895/how-do-i-get-the-directory-where-a-bash-script-is-located-from-within-the-script
 SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 # Cd into $FF_HOME. Assumes this script is in $FF_HOME/docker
 cd "$SCRIPT_DIR/.."
 
+#!/bin/bash
+
+# Default values
+cuda_version=$(nvcc --version | grep "release" | awk '{print $NF}')
+# Change cuda_version eg. V11.7.99 to 11.7
+cuda_version=${cuda_version:1:4}
+image="flexflow"
+
+# Parse command-line options
+while [[ $# -gt 0 ]]; do
+  key="$1"
+
+  case $key in
+    --cuda_version)
+      cuda_version="$2"
+      shift 2
+      ;;
+    --image_name)
+      image="$2"
+      shift 2
+      ;;
+    *)
+      echo "Invalid option: $key"
+      exit 1
+      ;;
+  esac
+done
+
+if [[ "$cuda_version" != @(11.1|11.3|11.5|11.6|11.7|11.8) ]]; then
+  # validate the verison of CUDA against a list of supported ones
+  # 11.1, 11.3, 11.5, 11.6, 11.7, 11.8
+  echo "cuda_version is not supported, please choose among {11.1,11.3,11.5,11.6,11.7,11.8}"
+  exit 1
+fi
+
+# modify cuda version to xx.x.0
+cuda_version=${cuda_version}.0
```

Comment:
here, we have to be a bit careful because nvidia does not distribute docker images versioned xx.x.0 for some CUDA versions. For example, for CUDA 11.7, only 11.7.1 is available, but not 11.7.0. Same for 11.1. For 11.6, only 11.6.2 is available, but not 11.6.0 nor 11.6.1. So we might need to have a different default sub-sub-version for each subversion of CUDA